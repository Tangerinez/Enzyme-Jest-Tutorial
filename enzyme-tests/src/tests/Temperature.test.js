import Temperature from "../components/temperature/temperature";
import React from "react";
import sinon from "sinon";
import { shallow, mount } from "enzyme";

it("renders correctly", () => {
  const wrapper = shallow(
    // light-weight representation of rendered component
    <Temperature temp={10} city="Toronto" toggleForecast={() => {}} />
  );
  expect(wrapper).toMatchSnapshot();
});

it("formats temp correctly", () => {
  // mount our Temperature component
  const wrapper = mount(
    <Temperature temp={10} city="Toronto" toggleForecast={() => {}} />
  );

  // extract the text from the LargeText styled component
  const text = wrapper.find("p").text();

  // ensure it matches what is expected
  expect(text).toEqual("10Â°c");
});

it("calls toggleForecast on click", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <Temperature temp={10} city="Toronto" toggleForecast={spy} />
  );

  // find the first div and simulate a click event on it
  wrapper
    .find("div")
    .first()
    .simulate("click");

  // ensure that our spy (toggleForecast) was called when click was simulated
  expect(spy.calledOnce).toBe(true);
});
