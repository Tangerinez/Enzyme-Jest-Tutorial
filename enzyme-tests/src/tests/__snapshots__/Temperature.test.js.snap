// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  onClick={[Function]}
>
  <p>
    10
    Â°c
  </p>
  <div>
    Toronto
  </div>
</div>
`;
